packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -93,26 +97,53 @@ export class DefaultCrudRepository<T extends Entity, ID>
   ) {
     const definition = entityClass.definition;
     assert(
-      !!definition,
-      `Entity ${entityClass.name} must have valid model definition.`,
+      definition.idProperties().length > 0,
+      `Entity ${entityClass.name} must have at least one id/pk property.`,
     );
 
+    this.setupModel(entityClass);
+    this.setModelClass();
+  }
+  private setupModel(entityClass: typeof Model & {prototype: T}) {
+    const definition = entityClass.definition;
+
     assert(
-      definition.idProperties().length > 0,
-      `Entity ${entityClass.name} must have at least one id/pk property.`,
+      !!definition,
```

Comment:
As I wrote above, this check should be run as the first one.

Reply:
This check also runs when it's a embedded model.